### ISSUE TITLE
Shift button does not work as expected

### ISSUE THREAD

**Issue by KES777** [linuxmint/linuxmint#113]
When I upgrade from 18.3 to 19.1 and use combinations of keys SHIFT + ...
For example change keyboard layout: SHIFT+ALT

shift is not taken into account when typing.

How to reproduce:
1. Change keyboard: press SHIFT+ALT (do not release)
2. Release ALT key, SHIFT is holded
3. press `:`, but instead you get `;`  (unexpected)
4. press `D`, but instead you get `d` (unexpected)
5. release SHIFT
6. press SHIFT+D, you get `D` Now it is as expected.

**Comment by KES777** [comment]
Also `ALT+SHIFT+TAB` does not work

**Comment by xenopeek** [comment]
I hope you managed to solve this issue. Tried to reproduce it on Linux Mint 19.3 Cinnamon with your steps but the issue doesn't happen here. Closing this here also considering its age. If you still have this issue you can reopen this issue. Please then add your current Linux Mint version and desktop environment.

**Comment by KES777** [comment]
To reopen should I create new one or just write here?

Problem still exists on Linux Mint 19.3

    Linux work 5.3.0-24-generic #26~18.04.2-Ubuntu SMP Tue Nov 26 12:34:22 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

**Comment by KES777** [comment]
Task back switching hot key also still does not work: `ALT+SHIFT+TAB`
I need to go over whole list of tasks with `ALT+TAB` to switch to previous task.
I need to go back over the list, for example, when I occasionally press excess `ALT+TAB`

**Comment by xenopeek** [comment]
I meant are you using Cinnamon, MATE or Xfce.

**Comment by KES777** [comment]
Cinammon

**Comment by xenopeek** [comment]
Did you mean in your first post with "For example change keyboard layout: SHIFT+ALT" that you set the keyboard shortcut for changing keyboard layout to just modifier keys shift+alt? As a test I set the keyboard shortcut to launch the terminal to shift+alt.

If I press alt first then shift, hold both down, then also press tab, a terminal opens after pressing down shift and the task list pops up when hitting tab and starts moving in reverse on hitting tab again. If I first press shift then alt, the terminal doesn't pop up and task list works as expected. If I press alt + shift + tab at exactly the same time then hold down alt + shift a terminal also doesn't pop up and task list works as expected.

This is the same across Linux Mint 19.3 Cinnamon and Xfce. MATE won't let users set a shortcut of just modifier keys.

I can't say if this worked differently on 18.3 but I think this should be expected behavior. I mean, when you set the shortcut for something to just modifier keys shift+alt, any keyboard shortcut using those modifier keys will also trigger that shortcut if you don't press the non-modifier key of the other keyboard shortcut at exactly the same time.

Several workarounds:

1. Press shift first, than alt; this doesn't trigger the shift+alt shortcut for me so should let you trigger the other shift+alt+non-modifier key shortcut.
2. Press shift+alt+non-modifier key at exactly the same time; this also doesn't trigger the shift+alt shortcut for me.
3. Don't use just modifier keys in shortcuts.

**Comment by KES777** [comment]
At 17.3 Cinnamon that works without any workaround.

**Comment by KES777** [comment]
ALT+SHIFT is standart hotkey (look at picture)

![_2](https://user-images.githubusercontent.com/799021/73120917-a716b800-3f7c-11ea-9ac5-5c5036454d1e.jpg)

2. Even more I can not assign ALT+SHIFT+TAB

![_](https://user-images.githubusercontent.com/799021/73120922-bd247880-3f7c-11ea-82b8-95638b66a2ee.jpg)

**Comment by xenopeek** [comment]
For point 1, I didn't know that option was there. Thanks for showing where to find it. With that I was also able to reproduce the issue on Linux Mint 19.3 MATE. So this doesn't look to be tied to desktop environment but probably X server (which is common across all three).

For point 2, that's also unassigned here in 19.3 Cinnamon. If I set the layout switch to Alt+Shift as you showed where to do the workaround I mentioned before doesn't work. Whether I press Alt+Shift at same time, Alt first then Shift or Shift then Alt, the layout is always switched regardless of at same time or afterwards pressing Tab.

I further tried to reproduce this on a different distro, using Gnome 3.34. Sorry to report but setting layout switcher to Alt+Shift there in the same layout options menu also break Shift+Alt+Tab and I can no longer cycle backwards through open windows with that set (unsetting a shortcut for layout switcher make cycle backwards work again).

Back to Linux Mint 19.3 Cinnamon: I could set a different shortcut for cycle backwards. I set cycle to Ctrl+Tab, cycle backwards to Shift+Ctrl+Tab and layout switch to Alt+Shift and that plays nice together. I get that you want it to work like it did on 17.3 but this is not something that can be fixed here, if it can be fixed at all. It not working as you say it did before is not due to change in desktop environment but some other underlying component. X server seems likely.
